### ISSUE TITLE
Dropdown opens then immeditately closes on first focus of single select on touch devices

### ISSUE THREAD

**Issue by coreycummings** [harvesthq/chosen#2171]
For iPad, The first time I focus on the drop down and click, it opens the list items and then automatically closes, next click it works and every click after that works. To reproduce, I can go to http://harvesthq.github.io/chosen/ and the first example has the issue when you select the drop down for the first time. Use chrome and open developer tools, click the phone icon and select iPad3/4 and refresh page, then select the first chosen example drop down, it closes the drop down on first selection during touchend. If I first focus on the element and then click or touch, then the problem does not occur. It seems like it has something to do with showing the drop down list and focusing on the element that causes it to collapse itself.

**Comment by MyDino** [comment]
I have a similiar issue on iPad and only if page scrolls to open keyboard, when select activated: one of the first 3 entries is selected automatically and loaded directly without user interaction.
I use chosen since years, this problem is new to me. How can I prevent selection through iPad keyboard popup?

**Comment by qwelyt** [comment]
We have this issue as well. The solutions I've found is to add `evt.preventDefault()` in `Chosen.prototype.container_mousedown = function(evt)`, like this: 
````javascript
Chosen.prototype.container_mousedown = function(evt) {
  if (!this.is_disabled) {
    if (evt && evt.type === "mousedown" && !this.results_showing) {
      evt.preventDefault();
    }
        
    evt.preventDefault(); // Disable the double-click behaviour for touch-devices.
    if (!((evt != null) && ($(evt.target)).hasClass("search-choice-close"))) {
````
However, this breaks multi-selects. You will have to click somewhere else on the page to be able to select another option. It does work for single-selects though.

The other option, and the one we are using, is to prevent any non-desktop browser to use Chosen. Which limits who can use the Chosen-alternative quite dramatically in our product.

**Comment by khayes-wdas** [comment]
We're having this issue as well. We went with _Chosen_ over _Select2_ because of better mobile support (primarily phone). It would be great to get the other touch devices as well. Frankly, for touch, it would be nice if we had the option to make iPads work like the iPhone does where it uses the native dropdown.

**Comment by kylehayes** [comment]
We're having this issue as well. We went with Chosen over Select2 because of better mobile support (primarily phone). It would be great to get the other touch devices as well. Frankly, for touch, it would be nice if we had the option to make iPads work like the iPhone does where it uses the native dropdown.

**Comment by MyDino** [comment]
@qwelyt: your solution works well to prevent auto-select, but unfortunately list scrolling is broken then in single select

**Comment by afattahi54** [comment]
As refereed above .... #2239 
same issue happens on Android devices too, I have tested on Samsung Tab. I wonder if it will going to be fixed.
PS: I am not a jQuery guy, during project start up I selected the chosen over select2 because of its high ranking in github (more stars !). It select 2 could work fine on touch devices, same could happen to chosen, I prefer to keep using it and looking for some fixes :)
